src/qibocal/cli/fit.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import json
+import pathlib
+
+import yaml
+from qibolab.serialize import dump_runcard
+
+from ..auto.execute import Executor
+from ..auto.history import add_timings_to_meta
+from ..auto.mode import ExecutionMode
+from ..auto.runcard import Runcard
+from .utils import META, RUNCARD, UPDATED_PLATFORM
+
+
+def run_fit(folder, update):
+    """Post-processing analysis
+
+    Arguments:
+
+    - FOLDER: input folder.
+
+    """
+    # load path, meta, runcard and executor
+    path = pathlib.Path(folder)
```

Comment:
```suggestion
def run_fit(folder: Path, update):
    """Post-processing analysis

    Arguments:

    - FOLDER: input folder.

    """
```

Try to avoid converting as much as possible: if you need a `Path`, ask for it.

At some point, you'll hit the user. But if it's in Python, you should also ask for a `Path`. In this case, it is in the CLI, but `click` can directly provide you a `Path` (and check it exists or not, if you need it).

Reply:
Also, dropping the conversion allows you to keep using `folder` as a name, even without shadowing it. And `folder` is more specific than `path` (slightly, but meaningfully).

Reply:
> In this case, it is in the CLI, but click can directly provide you a Path (and check it exists or not, if you need it).
Indeed this is correct. I've added the `[path_type` ](https://click.palletsprojects.com/en/8.1.x/parameters/).
Very useful!

Also from the discussion I forgot whether it is better to put `folder` or `path`. For now I converted everything to path.